mllam_data_prep/ops/cropping.py

```diff
@@ -0,0 +1,201 @@
+import numpy as np
+import spherical_geometry as sg
+import xarray as xr
+from spherical_geometry.polygon import SphericalPolygon
+
+
+def _get_latlon_coords(da: xr.DataArray) -> tuple:
+    """
+    Get the latlon coordinates of a DataArray.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        The data.
+
+    Returns
+    -------
+    tuple (xarray.DataArray, xarray.DataArray)
+        The latitude and longitude coordinates.
+    """
+    if "latitude" in da.coords and "longitude" in da.coords:
+        return (da.latitude, da.longitude)
+    elif "lat" in da.coords and "lon" in da.coords:
+        return (da.lat, da.lon)
+    else:
+        import ipdb
+
+        ipdb.set_trace()
+        raise Exception("Could not find lat/lon coordinates in DataArray.")
+
+
+def create_convex_hull_mask(ds: xr.Dataset, ds_reference: xr.Dataset) -> xr.DataArray:
+    """
+    Create a grid-point mask for lat/lon coordinates in `da` indicating which
+    points are interior to the convex hull of the lat/lon coordinates of
+    `da_ref`.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        The dataset for which to create the mask.
+    ds_reference : xarray.Dataset
+        The reference dataset from which to create the convex hull of the coordinates.
+
+    Returns
+    -------
+    xarray.DataArray
+        A boolean mask indicating which points are interior to the convex hull.
+    """
+    da_lat, da_lon = _get_latlon_coords(ds)
```

Comment:
When testing this out I get an error here. Now the example ERA5 config renames latitude+longitude to y+x, which I think is needed to make the output work with neural-lam? It does however mean that there are no lat/lon in the coords of the dataset here.

Reply:
Diff for local changes I had to do to get it running. With this I got rid of the errors and was able to save an ERA5 zarr.

```
--- a/mllam_data_prep/ops/cropping.py
+++ b/mllam_data_prep/ops/cropping.py
@@ -22,6 +22,8 @@ def _get_latlon_coords(da: xr.DataArray) -> tuple:
         return (da.latitude, da.longitude)
     elif "lat" in da.coords and "lon" in da.coords:
         return (da.lat, da.lon)
+    elif "y" in da.coords and "x" in da.coords:
+        return (da.y, da.x)
     else:
         import ipdb
 
@@ -132,9 +134,9 @@ def distance_to_convex_hull_boundary(
     # only consider points that are external to the convex hull
     ds_exterior = ds.where(~da_ch_mask, drop=True)
 
-    da_xyz = _latlon_to_unit_sphere_xyz(ds_exterior.lat, ds_exterior.lon)
+    da_xyz = _latlon_to_unit_sphere_xyz(ds_exterior.y, ds_exterior.x)
     da_xyz_ref = _latlon_to_unit_sphere_xyz(
-        ds_reference_separate_gridindex.lat, ds_reference_separate_gridindex.lon
+        ds_reference_separate_gridindex.y, ds_reference_separate_gridindex.x
     )
```

Reply:
Above makes the code run with the config, but the distance computation then breaks. I can't figure out why, but I think it relates to mixing of xy and latlon coordinates somewhere.

Reply:
Thanks! Assuming that the latitude and longitude values are available as `.lat` and `.lot` on the dataset was a mistake (I meant to wrap all the access to the latitude and longitude coordinates with the `_get_latlon_coords` function, but missed this one. If I had implemented the testing before I would have caught this...)

The code in neural-lam actually sets the cartesian coordinates (or maybe they should be called "regular grid coordinates") when loading the dataset: https://github.com/mllam/neural-lam/blob/main/neural_lam/datastore/mdp.py#L108 and uses the config to work out the name of the coordinates that were stacked. So the coordinates that were stacked don't have to be renamed to `x` and `y` (that is the idea anyway)